read.go

```diff
@@ -42,6 +42,28 @@ func readInt64(r *bufio.Reader, sz int, v *int64) (int, error) {
 	return peekRead(r, sz, 8, func(b []byte) { *v = makeInt64(b) })
 }
 
+func readVarInt(r *bufio.Reader, sz int, v *int64) (int, error) {
```

Comment:
I think that you can rewrite this function to make it more concise, reduce the number of loops, and also make it leverage/read more like kafka-go.  Something like:

```
func readVarInt(r *bufio.Reader, sz int, v *int64) (remain int, err error) {
	l := 0
	for done := false; !done && err == nil; {
		remain, err = peekRead(r, sz, 1, func(b []byte) {
			done = b[0]&0x80 == 0
			*v |= int64(b[0]&0x7f) << uint(l*7)
		})
		l++
	}
	return
}
```